### PR TITLE
test(prompts): add unit tests for prompt templates, context_cmd, and helpers

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Add **66 unit tests** covering previously untested areas of the `gptme/prompts/` module. This module (504 lines across 5 files) had zero dedicated unit tests for context command handling, template composition, chat history config, and agent file discovery.

### Test Coverage

| Test Class | Tests | What it covers |
|-----------|-------|----------------|
| TestTruncateContextOutput | 7 | `_truncate_context_output` — truncation logic, newline breaks, edge cases |
| TestGetProjectContextCmdOutput | 4 | `get_project_context_cmd_output` — success, failure, timeout, large output |
| TestJoinMessages | 5 | `_join_messages` — content joining, flag preservation, role validation |
| TestPromptGptme | 9 | `prompt_gptme` — agent_name, thinking tags, model display, base_prompt override |
| TestPromptComposition | 4 | `prompt_full`/`prompt_short` — section composition, interactive modes |
| TestPromptUser | 4 | `prompt_user` — config fallback between `[user]` and `[prompt]` sections |
| TestPromptProject | 3 | `prompt_project` — markdown mode, with/without project info |
| TestPromptSysteminfo | 3 | `prompt_systeminfo` — OS info, workspace path, format |
| TestPromptTimeinfo | 3 | `prompt_timeinfo` — date, markdown/XML format |
| TestUseChatHistoryContext | 10 | `use_chat_history_context` — env var flag parsing (truthy/falsy) |
| TestFindAgentFilesInTree | 6 | `find_agent_files_in_tree` — directory walking, exclusions, cross-tool |
| TestXmlSection | 3 | `_xml_section` — wrapping, whitespace, nested XML |
| TestConstants | 5 | `AGENT_FILES`, `DEFAULT_CONTEXT_FILES`, backward compat alias |

All 66 tests pass. No production code changes.

## Test plan

- [x] All 66 new tests pass locally
- [x] Existing prompt test suites still pass (41 tests)
- [x] mypy clean
- [ ] CI green

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: tests/test_prompt_templates.py [hunk 1]
```diff
@@ -0,0 +1,838 @@
...
+        # Result should end at a line boundary before the TRUNCATED notice
+        main_content = result.split("\n\n... [TRUNCATED")[0]
+        # Should not end mid-line
+        assert main_content.endswith(main_content.rstrip())
+
+    def test_truncation_notice_format(self):
+        """Truncation notice includes original and kept char counts."""
+        text = "a" * 500
+        result = self._truncate(text, 200)
+        assert "[TRUNCATED:" in result
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:69]: Fixed in b268a70 — tool now has examples so the assertion is meaningful.

File: tests/test_prompt_templates.py [hunk 2]
```diff
@@ -0,0 +1,841 @@
...
+
+        # Tool should be mentioned but without examples section
+        assert "test-tool" in combined
+        assert "### Examples" not in combined
+
+
+# ---------------------------------------------------------------------------
+# prompt_user
+# ---------------------------------------------------------------------------
+
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:466]: Fixed in b268a70 — ToolSpec now includes examples, assertion verifies they're excluded.

File: tests/test_prompt_templates.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:69]: Fixed — now verifies the last line is a complete 'line N' entry instead of the trivially-true endswith(rstrip()) pattern.

File: tests/test_prompt_templates.py [hunk 3]
```diff
@@ -0,0 +1,838 @@
...
+        msgs = list(prompt_gptme(interactive=True))
+        content = msgs[0].content
+        assert "interactive mode" in content
+        assert "non-interactive" not in content
+
+    def test_non_interactive_mode(self):
+        """Non-interactive mode includes non-interactive instructions."""
+        from gptme.prompts import prompt_gptme
+
+        msgs = list(prompt_gptme(interactive=False))
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:239]: Fixed — added autouse fixture that mocks both get_project_git_dir (returns None) and get_project_config (returns MagicMock with base_prompt=None), so tests are isolated from live repo config.

File: tests/test_prompt_templates.py [hunk 4]
```diff
@@ -0,0 +1,838 @@
...
+        mock_config.get_env.return_value = value
+
+        with patch("gptme.prompts.chat_history.get_config", return_value=mock_config):
+            assert use_chat_history_context() is False
+
+
+# ---------------------------------------------------------------------------
+# find_agent_files_in_tree
+# ---------------------------------------------------------------------------
+
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:736]: Good catch — renamed to 'other' and updated docstring to clarify the allowlist-based truthy check semantics.

File: tests/test_prompt_templates.py [hunk 5]
```diff
@@ -0,0 +1,862 @@
...
+# ---------------------------------------------------------------------------
+# prompt_user
+# ---------------------------------------------------------------------------
+
+
+class TestPromptUser:
+    """Tests for prompt_user in templates.py — config fallback logic."""
+
+    def test_default_user_prompt(self):
+        """Without config, uses default user description."""
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py]: Fixed — test_default_user_prompt now mocks get_config() with all fields set to None, verifying the built-in defaults ('# About User', 'You are interacting with a human programmer.', 'No specific preferences set.') are used when no config is present.

File: tests/test_prompt_templates.py [hunk 6]
```diff
@@ -0,0 +1,862 @@
...
+    def test_prompt_full_non_interactive_skips_user(self):
+        """Non-interactive mode should skip user preferences."""
+        from gptme.prompts.templates import prompt_full
+
+        tools = [self._make_tool()]
+        msgs = list(
+            prompt_full(
+                interactive=False,
+                tools=tools,
+                tool_format="markdown",
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:424]: Fixed — test_prompt_full_non_interactive_skips_user now also asserts '# About' and 'Response Preferences' are absent, verifying prompt_user() was not called (not just that a non-interactive blurb is present).

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: tests/test_prompt_templates.py [hunk 7]
```diff
@@ -0,0 +1,875 @@
...
+
+    def test_prompt_full_includes_all_sections(self):
+        """prompt_full should include gptme, tools, user, project, systeminfo, timeinfo."""
+        from gptme.prompts.templates import prompt_full
+
+        tools = [self._make_tool()]
+        msgs = list(
+            prompt_full(
+                interactive=True,
+                tools=tools,
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:367]: Fixed in f6f5209 — added the same `_isolate_from_live_config` autouse fixture to `TestPromptComposition` that already exists in `TestPromptGptme`.

File: tests/test_prompt_templates.py [hunk 8]
```diff
@@ -0,0 +1,875 @@
...
+            msgs = list(prompt_project())
+
+        assert len(msgs) == 1
+        assert "bare-project" in msgs[0].content
+
+
+# ---------------------------------------------------------------------------
+# prompt_systeminfo
+# ---------------------------------------------------------------------------
+
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:620]: Fixed in f6f5209 — guarded the markdown path against `None` project_info (the XML path already had this guard), and added `assert "None" not in msgs[0].content` to the test to lock in the fix.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: tests/test_prompt_templates.py [hunk 9]
```diff
@@ -0,0 +1,889 @@
...
+        with patch("gptme.prompts.templates.get_project_git_dir", return_value=None):
+            msgs = list(prompt_project())
+        assert len(msgs) == 0
+
+    def test_project_with_info(self):
+        """Project with info shows project name and description."""
+        from gptme.prompts import prompt_project
+
+        project_dir = Path("/fake/my-project")
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_prompt_templates.py:566]: Fixed in da94ad5 — added  mock to  with explicit user fields (name, about, response_preference), matching the pattern used by the other three tests in .
